src/components/body/body.component.ts

```diff
@@ -452,7 +451,7 @@ export class DataTableBodyComponent implements OnInit, OnDestroy {
    */
   getRowAndDetailHeight(row: any): number {
     let rowHeight = this.getRowHeight(row);
-    const expanded = this.rowExpansions.get(row);
+    const expanded = this.rowExpansions.get(row.key);
```

Comment:
adding a dependency on an arbitrarily named field seems fragile and non-obvious. what if you used the rowIdentifier function to get the key?

Reply:
hey mark, thanks for taking a look at this PR.  getRowAndDetailHeight is called by getGroupHeight which is actually passing not a regular row, but a 'grouped row' which is created by groupArrayBy in datatable.component.cs.  That function creates a may with key/value pairs, where 'key' has the value of the unique set of values defined by the 'groupByRows' property, and value is the list of rows that match that key.  So in this case, key always exists and it intentional to query the rowExpansions by key as this is the way they are grouped.  Hope that makes sense!  I did just notice in this commit I accidentally included an fdescribe in the tests, will remove that now.

Reply:
@markwaddle hi mark do you need any further clarification, would be great to get this PR merged if you are happy with it.